src/utils/ProfileUtils.ts

```diff
@@ -201,17 +201,23 @@ export async function populateImportedProfile(
     }
 }
 
-//TODO: Check if instead of returning null/empty strings, there's some errors that should be handled
-export async function readProfileFile(file: string) {
-    let read = '';
+export async function readProfileFile(file: string): Promise<string> {
+    let read: string | null | undefined;
     if (file.endsWith('.r2x')) {
         read = (await FsProvider.instance.readFile(file)).toString();
     } else if (file.endsWith('.r2z')) {
-        const result: Buffer | null = await ZipProvider.instance.readFile(file, "export.r2x");
-        if (result === null) {
-            return null;
-        }
-        read = result.toString();
+        ZipProvider.instance.readFile(file, "export.r2x").then(
+            (value) => {
+                read = value ? value.toString() : null;
+            },
+        );
+    }
+
+    if (!read) {
```

Comment:
This approach doesn't work for me at all. At this point the callback hasn't fired and `read` is always `undefined` for valid r2x files.